src/DefaultBuilder/src/WebHost.cs

```diff
@@ -224,6 +229,20 @@ internal static void ConfigureWebDefaults(IWebHostBuilder builder)
 
                 services.AddTransient<IStartupFilter, HostFilteringStartupFilter>();
 
+                if (string.Equals("true", hostingContext.Configuration["ForwardedHeaders_Enabled"], StringComparison.OrdinalIgnoreCase))
```

Comment:
We might want to consolidate these setting strings somewhere instead.

Reply:
Can you do some performance tests? More default middleware in the request pipeline means it needs to be scrutinized for performance and allocations...

Reply:
@davidfowl I'll file a separate task. Any improvements to the middleware itself would not overlap with the changes in this PR.

Reply:
https://github.com/aspnet/AspNetCore/issues/10276

Reply:
Dude you can’t introduce allocations into the mainline. As long as you know you’re not doing that I’ll be fine. I’ll be very unhappy otherwise

Reply:
? This isn't the first time we've added this middleware, it has been added by default for IIS out-of-proc since 1.0. This change only adds it when you publish to Azure Linux containers.

On a perf note the middleware is [synchronous](https://github.com/aspnet/AspNetCore/blob/1dbc203e197eaa2dc7f1e243ed74cf3ca03efcfb/src/Middleware/HttpOverrides/src/ForwardedHeadersMiddleware.cs#L140-L144), it does not add an async state machine. Its primary allocations are for header [splitting](https://github.com/aspnet/AspNetCore/blob/1dbc203e197eaa2dc7f1e243ed74cf3ca03efcfb/src/Middleware/HttpOverrides/src/ForwardedHeadersMiddleware.cs#L158), and most of the time there will only be a single header entry so no allocation will be required there. Eventually it allocates one IPAddress and one IPEndPoint.

Reply:
Will it only allocate if those headers exist?

Reply:
So I did the work to profile and it allocates even if there's nothing to do:

![image](https://user-images.githubusercontent.com/95136/57857829-c54cd700-77a4-11e9-937a-a20b1ae50654.png)

Please fix it so that it doesn't allocate unless it *has* to.

Reply:
> it allocates even if there's nothing to do

In the scenarios where it's enabled, there's always something to do. These headers are not optional in these environments. For a given deployment all requests are expected to have similar headers.

Reply:
I don't have many hills but this one I'm willing to die on. The needless allocations need to be fixed.

Reply:
To be clear, right now this only goes on by default if you explicitly turn it on with an environment variable. The only place we set that is in App Service for Linux which has a front-end that will always provide these headers.

@davidfowl so you want to block this PR on this work rather than handling it inhttps://github.com/aspnet/AspNetCore/issues/10276 ? Just want to make sure we're all on the same page. I do agree that we need to do the work when we introduce these new defaults. My main concern is understanding the true "do nothing" scenario here. If the environment variable is set, the middleware should _never_ "do nothing".

Reply:
Spoke to Chris, I didn't realize it wasn't on by default. We did discuss that using the middleware without the header is "invalid" yet it just passes through and could maybe fix that. No need to block the PR anymore.

Reply:
Resolved offline. He didn't understand when it would be enabled.